tests/expressions_window/tests.py

```diff
@@ -0,0 +1,755 @@
+import datetime
+from unittest import skipIf, skipUnless
+
+from django.core.exceptions import FieldError
+from django.db import NotSupportedError, connection
+from django.db.models import F, RowRange, ValueRange, Window
+from django.db.models.aggregates import Avg, Max, Min, Sum
+from django.db.models.expressions import ExpressionList, OrderBy, Value
+from django.db.models.functions import (
+    CumeDist, DenseRank, ExtractYear, FirstValue, Lag, LastValue, Lead,
+    NthValue, Ntile, PercentRank, Rank, RowNumber,
+)
+from django.test import TestCase, skipUnlessDBFeature
+
+from .models import WindowTestModel
+
+
+@skipUnlessDBFeature('supports_over_clause')
+class TestWindowFunction(TestCase):
```

Comment:
There are a few window functions which take an `offset` argument, and thus at least one department with more than 2 employees is needed. I'm also a bit uncertain if all databases compute the same `ntile` if the number of rows in a partition does not divide the number of buckets evenly.

Also, tests for frames (which should be cleaned up) require a reasonable number of rows to make sense.